File: Order-service/src/main/java/org/model/order/service/customer/api/CustomerService.java
package org.model.order.service.customer.api;


import org.model.order.dto.customer.GetCustomerResponse;
import org.model.order.dto.customer.PutCustomerRequest;

import java.util.List;
import java.util.UUID;

public interface CustomerService {

    void addCustomer(GetCustomerResponse getCustomerResponse);

    void deleteCustomer(UUID id);

    List<GetCustomerResponse> getAllCustomers();

    GetCustomerResponse updateCustomer(UUID id, PutCustomerRequest putCustomerRequest);

    GetCustomerResponse getCustomerById(UUID id);
}

File: spring-angular/src/app/customer/service/customer.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {map, mergeMap, Observable} from "rxjs";
import {CustomerDetails} from "../model/customer-details";
import {Customers} from "../model/customers";
import {CustomerForm} from "../model/customer-form";
import {Orders} from "../../order/model/orders";

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http: HttpClient) {

  }

  //get all customers
  getCustomers(): Observable<Customers> {
    return this.http.get<Customers>('/api/customers');
  }

  //get a customer
  getCustomer(uuid: string): Observable<CustomerDetails> {
    return this.http.get<CustomerDetails>('/api/customers/' + uuid).pipe(
        mergeMap((customer: CustomerDetails) => {
          return this.http.get<Orders>('api/orders/', {params: {customer_id: uuid}}).pipe(
              map((orders: Orders) => {
                customer.orders = orders;
                return customer;
              })
        );
        })
    );
  }

  //delete customer
  deleteCustomer(uuid: string): Observable<any> {
    return this.http.delete('/api/customers/' + uuid);
  }

  //create customer
  postCustomer(request: CustomerForm): Observable<any> {
    return this.http.post('/api/customers', request);
  }

  //update customer
  putCustomer(uuid: string, request: CustomerForm): Observable<any>{
    return this.http.put('/api/customers/' + uuid, request);
  }

}
